cartridges/custom_cartridge/cartridge/controllers/CheckoutServices.js

```diff
@@ -0,0 +1,594 @@
+        });
+        return next();
+    }
+
+    // Handles payment authorization
+    var handlePaymentResult = COHelpers.handlePayments(order, order.orderNo);
+
+    // Handle custom processing post authorization
+    var options = {
+        req: req,
+        res: res
+    };
+    var postAuthCustomizations = hooksHelper('app.post.auth', 'postAuthorization', handlePaymentResult, order, options, require('*/cartridge/scripts/hooks/postAuthorizationHandling').postAuthorization);
+    if (postAuthCustomizations && Object.prototype.hasOwnProperty.call(postAuthCustomizations, 'error')) {
+        res.json(postAuthCustomizations);
+        return next();
+    }
+
+    if (handlePaymentResult.error) {
+        res.json({
+            error: true,
+            errorMessage: Resource.msg('error.technical', 'checkout', null)
+        });
+        return next();
+    }
+
+    var fraudDetectionStatus = hooksHelper('app.fraud.detection', 'fraudDetection', currentBasket, require('*/cartridge/scripts/hooks/fraudDetection').fraudDetection);
+    if (fraudDetectionStatus.status === 'fail') {
+        Transaction.wrap(function () { OrderMgr.failOrder(order, true); });
+
+        // fraud detection failed
+        req.session.privacyCache.set('fraudDetectionStatus', true);
+
+        res.json({
+            error: true,
+            cartError: true,
+            redirectUrl: URLUtils.url('Error-ErrorCode', 'err', fraudDetectionStatus.errorCode).toString(),
+            errorMessage: Resource.msg('error.technical', 'checkout', null)
+        });
+
+        return next();
+    }
+
+    // Places the order
+    var placeOrderResult = COHelpers.placeOrder(order, fraudDetectionStatus);
+    if (placeOrderResult.error) {
+        res.json({
+            error: true,
+            errorMessage: Resource.msg('error.technical', 'checkout', null)
+        });
+        return next();
+    }
+
+    if (req.currentCustomer.addressBook) {
+
+        // save all used shipping addresses to address book of the logged in customer
+        var allAddresses = addressHelpers.gatherShippingAddresses(order);
+        allAddresses.forEach(function (address) {
+            if (!addressHelpers.checkIfAddressStored(address, req.currentCustomer.addressBook.addresses)) {
+
+                // save all used shipping addresses to address book of external cervice
+                var UUIDUtils = require('dw/util/UUIDUtils');
+                var integrationAddressId = UUIDUtils.createUUID();
+                var CustomerMgr = require('dw/customer/CustomerMgr');
+                var customer = CustomerMgr.getCustomerByCustomerNumber(
+                    req.currentCustomer.profile.customerNo
+                );
+                var profile = customer.getProfile();
+                var result;
+
+                var HookManager = require('dw/system/HookMgr');
+                if (HookManager.hasHook('app.register.requestCustomerToExternalService')) {
```

Comment:
You can move this logic in the addressHelper rather than replacing place order endpoint

Reply:
fixed